### ISSUE TITLE
Featurerequest - Making an endertank "creative"

### ISSUE THREAD

**Issue by drz2k** [Chicken-Bones/EnderStorage#64]
Hey,

could you see this coming in ? A creative version of endertank that has unlimited amount of its liquid for server adminstrators/modpack creators?

Cheers!

**Comment by Chicken-Bones** [comment]
don't other mods have infinite tanks that you could just feed into the endertank?

**Comment by drz2k** [comment]
Yes of course there quite some. But since I know that many serveradmins distribute it via an endertank it would cut a step and add to your mod :)

Why fill an endertank with something from an infinite tank when you can do it straight with the endertank? :dancer: 

Also there are some modpacks, that dont really have infinity tanks or mechanisms to fill the "administrativly" like horizons lava.

Im no developer but i think you could easily integrate this and make a good feature out of it :)

**Comment by Ecu** [comment]
I suppose I don't see any reason not to do this, but honestly...adding a block to do it specifically when a server administrator could just use one of the many mods that already offer this functionality...seems silly.

**Comment by drz2k** [comment]
Well sometimes you have modpacks that simply dont offer infinite sources. I know of servers that have to rely on staff refilling tanks or using some abstract and complex blood magic setups for refilling.

Horizons daybreaker is a good example for no infinite sources. 

Another example : The modpack im currently working on will have some kind of infinite blood source, but it also will require a more complex setup involving technomancy.

Ofc you are right when theres a mod that offers an infinite tank you could simply hook up with some pipes to the endertank, but its not allways the case :)

**Comment by Slind14** [comment]
:+1: not every pack has infinite tanks, e.g. Horizons Daybreaker

**Comment by Ecu** [comment]
Again guys...this is a server thing.  There are multiple ways this could be done without adding a new block (one of which is just a command block that re-fills the tank.  

That all said, I don't really see a problem really with adding a creative block for it.

I'd like to also state that, to be fair, servers giving out infinite liquid in the first place kind of ruins the point of it all anyways.  Tanks are meant to help you transport resources, not give everyone infinite resources.

**Comment by Slind14** [comment]
```
I'd like to also state that, to be fair, servers giving out infinite liquid in the first place kind of ruins the point of it all anyways. Tanks are meant to help you transport resources, not give everyone infinite resources.
```
The way we do it is, that the ender tank with a private frequency for infinite lava requires a ender-pump (or similar) to be crafted. Once the player has the ender-pump he would have infinite lava anyway. But having the tank avoids the heavy light updates whenever a lava source block is pumped.

**Comment by Ecu** [comment]
What about just having a config for infinite lava then?  Once you fill a take full of lava, it'll never run out?  That way a new block doesn't need to exist for no reason and still requires a small amount of initial investment.

Seem reasonable?

**Comment by Slind14** [comment]
https://i.imgur.com/jM49uHc.png
This is how it looks like, the first one is a normal ender tank and the second one has a diamond lock. This way you can use the normal tank like always but also have access to infinite lava once you are able to craft the ender-pump. 
If the config option could be per tank. Like looking at the tank and `/enderstorage infinite` that would work. But globally would prevent the user for other things and if use-able with other liquids,well free liquid ores.. :)

**Comment by Ecu** [comment]
If you actually read what I said...I said a config for infinite **lava** not infinite everything.  I'm aware of what servers use to create infinite lava systems already.

I don't approve of adding blocks needlessly to this mod.  I honestly feel that if a server wants infinite lava, they should use another mod for it.  However, if it is only so much as a config option that a full tank of **lava** becomes infinite, then that is fine.  It also means there need not be a central tank as anyone capable of filling a tank with **lava** would be able to create an source of it.

**Comment by Slind14** [comment]
ahh, I did understand you wrong. Its nothing pressuring for us. I just came across this and saw that we would probably use it. Atm, we use nbt edit blocks depending on the pack (increasing the volume of tanks to 2 billion), those don't need to be refilled often. They are connected to the private ender tank with the expensive public recipe. If players need more than one, they need to craft an expensive ender quarry again.

**Comment by Ecu** [comment]
I think I'll stick with this as a final acceptable solution.  @Chicken-Bones, if this works for you...we should go ahead with something alone these lines...

```json
infinite_liquids:  [
     { key: [ "red", "red", "red" ], liquid: "minecraft:lava" },
     { key: [ "blue", "blue", "blue" ], liquid: "minecraft:water" }
]
```

**Comment by Slind14** [comment]
Some sort of separation from a normal colored ender tank would be good. Else it would be super cheap to get or the non infinite ender tank super expensive.

**Comment by Ecu** [comment]
Tough?  If you're going to allow for infinite liquids of any type, its an all or nothing thing in my book.

Another suggestion was to allow for locking colors of Ender Chests/Tanks, which I approved of.  Which means, if you needed to have infinite tanks that people have to specifically craft, lock the frequencies and make certain ones cost more.

**Comment by Slind14** [comment]
If that works, it sounds possible and usable.

**Comment by Ecu** [comment]
It doesn't work yet, of course.  But, it is how I would suggest handling things.
